src/engine/framework/BaseCommands.cpp

```diff
@@ -650,10 +653,45 @@ namespace Cmd {
                 const std::string& name = isNamed ? args.Argv(1) : "";
                 const std::string& command = args.EscapedArgs(2 + +isNamed);
                 std::string delay = args.Argv(1 + +isNamed);
-                bool frames = delay.back() == 'f';
 
-                if (frames) {
-                    delay.erase(--delay.end()); //FIXME-gcc-4.6 delay.pop_back()
+                const char front = delay.front();
+
+                if (front < '0' || front > '9') {
+                    Print("delay: missing delay");
+                    return;
+                }
+
+                enum class delayUnit {
+                    MILLISECOND = 0,
+                    SECOND = 's',
+                    MINUTE = 'm',
+                    HOUR = 'h',
+                    FRAME = 'f',
+                };
+
+                delayUnit unit = delayUnit::MILLISECOND;
+                char back = delay.back();
+
+                switch(delayUnit(back)) {
+                    case delayUnit::SECOND:
+                    case delayUnit::MINUTE:
+                    case delayUnit::HOUR:
+                    case delayUnit::FRAME:
+                        unit = delayUnit(back);
+                        delay.erase(--delay.end()); //FIXME-gcc-4.6 delay.pop_back()
+                        back = delay.back();
+                        if (back == 'm' && unit == delayUnit::SECOND)
```

Comment:
How about `Str::IsSuffix` here

Reply:
I agree with slipher that this would be more readable.

Reply:
I don't think it would be more readable, at this point the `s` of `ms` is already stripped out. We can do `Str::IsSuffix` before stripping it anyway, but in all cases that still look weird to me when half the code uses one way to do it and other half doing it another way. Here we just do “we already detected unit is second, we now look at multiplier”.